UI/Specs/UI-team-lead-spec.md

```diff
@@ -0,0 +1,328 @@
+- getUserDetails API of the IUser interface is called to send the data to processing module.
+- sendDetailsToContent of contentIcommunicator is called to send data to content module.
+
+**2. Join Session :**
+- Asks for the boardID , user name,ip address and profile picture.
+- Provides an buon to Start Session.
+- Sends the boardID, ip address and username to the processing module.
+- Sends the boardID,ip address username and profile picture to the content module.
+- Create a new plain sheet with all the menu options.
+
+```java
+public interface IUser {
+    String getUserDetails(String userName, String ipAddress, String boardId);
+    String getUser(ArrayList<Position>);
+    void subscribeForChanges(String identifier,IChanges handler);
+}
+```
+
+**3. Brush :**
+- Brush is used to draw on the canvas.
+- When we click on brush we ask the client to provide brush color and the brush size.
+- When the brush is used to draw on the canvas we will calculate the pixels which are changed and store it in a array.
+- When is the mouse drag event ends we call draw_curve API of the object of type IDrawErase and send the array of pixels.
+
+**4. Eraser :**
+- Eraser is used to clean drawings on the canvas.
+- Eraser supports different sizes similar to brush size.
+- When the eraser is used to clean the canvas we will calculate the pixels which are changed and store it in a array.
+- When is the mouse drag event ends we call erase API of the object of type IDrawErase and send the array of pixels.
+
+```java
+public interface IDrawErase {
+    // parameter: ArrayList<Pixel> - an ArrayList of pixel with RGB values for random curve object 
+    void draw_curve(ArrayList <Pixel>);
+    // parameter: ArrayList<Position> - an ArrayList of pixel position for erase object
+    void erase(ArrayList <Position>);
+}
+```
+**5. Shapes :**
+- On click provides an menu of standard shapes like circle, rectangle, square and line segment.
+- Also provides the option to select the colour of the shape.
+``` java
+public interface IDrawShapes {
+    // parameter: Pixel center - center position with intensity for circle
+    //            radius - radius of circle
+    void draw_circle(Pixel center, float radius);
+    // makes an object for square parallel to the whiteBoard 
+    // parameter: Pixel start - start position for square with RGB values in whiteBoard
+    //            float length - length of the sqaure
+    void draw_square(Pixel start, float length);         
+    // parameter: Pixel start - top left position with RGB values for rectangle in whiteBoard
+    //            Pixel end - bottom right position with RGB values for rectangle in whiteBoard
+    void draw_rectangle(Pixel start, Pixel end); 
+    // makes an object for a line segment parallel to the whiteBoard 
+    // parameter: Pixel start - start position with RGB values
+    //            Pixel end - end position with RGB values
+    void draw_line(Pixel start, Pixel end);        
+}
```

Comment:
Please look at the interface [https://github.com/convict-git/KoraKaagaz/tree/master/Specs/Processing](https://github.com/convict-git/KoraKaagaz/tree/master/Specs/Processing) and make the changes.